src/test/java/parser/JsonParserTest.java

```diff
@@ -0,0 +1,125 @@
+package parser;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import shop.Cart;
+import shop.RealItem;
+import shop.VirtualItem;
+
+import java.io.*;
+import java.lang.reflect.Field;
+import java.util.List;
+
+public class JsonParserTest {
+    private JsonParser jsonParser;
+    private Cart cart;
+    private String cartName;
+    private File file;
+    private RealItem ri;
+    private VirtualItem vi;
+
+    @BeforeEach
+    void prepare() {
+        jsonParser = new JsonParser();
+        cartName = RandomStringUtils.random(10, true, false);
+        cart = new Cart(cartName);
+
+        ri = new RealItem();
+        ri.setName("RealItem");
+        ri.setPrice(10.75);
+        ri.setWeight(5.20);
+
+        vi = new VirtualItem();
+        vi.setName("VirtualItem");
+        vi.setPrice(30.43);
+        vi.setSizeOnDisk(53.98);
+
+        cart.addRealItem(ri);
+        cart.addVirtualItem(vi);
+
+        jsonParser.writeToFile(cart);
+        file = new File("src/main/resources/" + cartName + ".json");
+    }
+
+    @Test
+    void writeToFileWritesCorrectCartInfo() {
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            Assertions.assertEquals(reader.readLine(), String.format(
+                    "{\"cartName\":\"%s\",\"realItems\":[{\"weight\":%.1f,\"name\":\"%s\",\"price\":%.2f}]," +
+                            "\"virtualItems\":[{\"sizeOnDisk\":%.2f,\"name\":\"%s\",\"price\":%.2f}],\"total\":%.3f}",
+                    cartName, ri.getWeight(), ri.getName(), ri.getPrice(),
+                    vi.getSizeOnDisk(), vi.getName(), vi.getPrice(), cart.getTotalPrice()
+            ));
+        } catch (FileNotFoundException ex) {
+            throw new NoSuchFileException(String.format("File %s.json not found!", file), ex);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    void readFromFileReadsCorrectCartInfo() {
```

Comment:
As an option you can use reflection to read file :)
The first option (with BufferedReader ) looks a bit easier. 
But the same code of reading the file should me moved to utility class